### ISSUE TITLE
Kong 1.0.3 to 1.2.1 Migrations Oddity "aborted: another node is already performing database changes"

### ISSUE THREAD

**Issue by jeremyjpj0916** [Kong/kong#4849]
### Summary

Upgraded from 1.0.3 to Kong 1.2.1, seems fine but one event I find strange is that I always see this error message now on 1.2.1 with various migration commands:

```
/ $ kong migrations list
Executed migrations:
         core: 000_base, 001_14_to_15, 002_15_to_1, 003_100_to_110, 004_110_to_120
          jwt: 000_base_jwt, 001_14_to_15
          acl: 000_base_acl, 001_14_to_15
rate-limiting: 000_base_rate_limiting, 001_14_to_15, 002_15_to_10, 003_10_to_112
       oauth2: 000_base_oauth2, 001_14_to_15, 002_15_to_10
/ $ kong migrations finish
No pending migrations to finish
aborted: another node is performing database changes
/ $ kong migrations up
Database is already up-to-date
aborted: another node is performing database changes
/ $
```

Note the aborted: another node is already performing database changes message. Where and why do I get this. Is there a db table/element I can look into to give more insight?

Obviously comes from here: https://github.com/Kong/kong/blob/19f4fe64c7278ce57696f3600b204744d07b5a5d/kong/cmd/utils/migrations.lua#L8

So how do I become a leader? ✊ 😄 

### Steps To Reproduce

1. Be on Kong 1.0.3 , upgrade to 1.2.1 and everything seem smooth
2. But see that random error message in both my datacenters for no reason. Figured maybe it was a consistency thing across all C* nodes keyspaces but it still persists with that message regardless so its not that. As far as I can tell Kong is running as expected even when I --vv the finish we can see no issues too:

```
/ $ kong migrations finish --vv
2019/07/29 18:30:51 [verbose] Kong: 1.2.1
2019/07/29 18:30:51 [debug] ngx_lua: 10013
2019/07/29 18:30:51 [debug] nginx: 1013006
2019/07/29 18:30:51 [debug] Lua: LuaJIT 2.1.0-beta3
2019/07/29 18:30:51 [verbose] no config file found at /etc/kong/kong.conf
2019/07/29 18:30:51 [verbose] no config file found at /etc/kong.conf
2019/07/29 18:30:51 [verbose] no config file, skip loading
2019/07/29 18:30:51 [debug] reading environment variables
2019/07/29 18:30:51 [debug] KONG_ERROR_DEFAULT_TYPE ENV found with "text/plain"
2019/07/29 18:30:51 [debug] KONG_ADMIN_LISTEN ENV found with "0.0.0.0:8001"
2019/07/29 18:30:51 [debug] KONG_PROXY_ACCESS_LOG ENV found with "off"
2019/07/29 18:30:51 [debug] KONG_TRUSTED_IPS ENV found with "0.0.0.0/0,::/0"
2019/07/29 18:30:51 [debug] KONG_CASSANDRA_USERNAME ENV found with "dba_user"
2019/07/29 18:30:51 [debug] KONG_SSL_CERT_KEY ENV found with "/usr/local/kong/ssl/kongprivatekey.key"
2019/07/29 18:30:51 [debug] KONG_MEM_CACHE_SIZE ENV found with "1024m"
2019/07/29 18:30:51 [debug] KONG_CASSANDRA_PASSWORD ENV found with "******"
2019/07/29 18:30:51 [debug] KONG_PROXY_LISTEN ENV found with "0.0.0.0:8000, 0.0.0.0:8443 ssl http2"
2019/07/29 18:30:51 [debug] KONG_CASSANDRA_SSL ENV found with "on"
2019/07/29 18:30:51 [debug] KONG_DB_UPDATE_FREQUENCY ENV found with "5"
2019/07/29 18:30:51 [debug] KONG_DB_UPDATE_PROPAGATION ENV found with "5"
2019/07/29 18:30:51 [debug] KONG_CASSANDRA_PORT ENV found with "9042"
2019/07/29 18:30:51 [debug] KONG_DNS_ORDER ENV found with "LAST,SRV,A,CNAME"
2019/07/29 18:30:51 [debug] KONG_DNS_ERROR_TTL ENV found with "1"
2019/07/29 18:30:51 [debug] KONG_HEADERS ENV found with "latency_tokens"
2019/07/29 18:30:51 [debug] KONG_DNS_STALE_TTL ENV found with "4"
2019/07/29 18:30:51 [debug] KONG_DATABASE ENV found with "cassandra"
2019/07/29 18:30:51 [debug] KONG_UPSTREAM_KEEPALIVE ENV found with "60"
2019/07/29 18:30:51 [debug] KONG_ADMIN_ACCESS_LOG ENV found with "off"
2019/07/29 18:30:51 [debug] KONG_DB_CACHE_WARMUP_ENTITIES ENV found with "services,plugins,consumers"
2019/07/29 18:30:51 [debug] KONG_LUA_SOCKET_POOL_SIZE ENV found with "30"
2019/07/29 18:30:51 [debug] KONG_CASSANDRA_LOCAL_DATACENTER ENV found with "DC2"
2019/07/29 18:30:51 [debug] KONG_CASSANDRA_CONSISTENCY ENV found with "LOCAL_QUORUM"
2019/07/29 18:30:51 [debug] KONG_CLIENT_MAX_BODY_SIZE ENV found with "100m"
2019/07/29 18:30:51 [debug] KONG_ADMIN_ERROR_LOG ENV found with "/dev/stderr"
2019/07/29 18:30:51 [debug] KONG_DNS_NOT_FOUND_TTL ENV found with "30"
2019/07/29 18:30:51 [debug] KONG_PG_SSL ENV found with "off"
2019/07/29 18:30:51 [debug] KONG_REAL_IP_HEADER ENV found with "X-Forwarded-For"
2019/07/29 18:30:51 [debug] KONG_CASSANDRA_CONTACT_POINTS ENV found with "server8173,server8172,server8175,server8171,server8176,server8180"
2019/07/29 18:30:51 [debug] KONG_DNS_HOSTSFILE ENV found with "/etc/hosts"
2019/07/29 18:30:51 [debug] KONG_SSL_CERT ENV found with "/usr/local/kong/ssl/kongcert.crt"
2019/07/29 18:30:51 [debug] KONG_LUA_SSL_VERIFY_DEPTH ENV found with "3"
2019/07/29 18:30:51 [debug] KONG_CASSANDRA_TIMEOUT ENV found with "5000"
2019/07/29 18:30:51 [debug] KONG_CASSANDRA_SSL_VERIFY ENV found with "on"
2019/07/29 18:30:51 [debug] KONG_LOG_LEVEL ENV found with "notice"
2019/07/29 18:30:51 [debug] KONG_ANONYMOUS_REPORTS ENV found with "off"
2019/07/29 18:30:51 [debug] KONG_REAL_IP_RECURSIVE ENV found with "on"
2019/07/29 18:30:51 [debug] KONG_ROUTER_CONSISTENCY ENV found with "eventual"
2019/07/29 18:30:51 [debug] KONG_PROXY_ERROR_LOG ENV found with "/dev/stderr"
2019/07/29 18:30:51 [debug] KONG_CLIENT_BODY_BUFFER_SIZE ENV found with "50m"
2019/07/29 18:30:51 [debug] KONG_CASSANDRA_LB_POLICY ENV found with "RequestDCAwareRoundRobin"
2019/07/29 18:30:51 [debug] KONG_LUA_SSL_TRUSTED_CERTIFICATE ENV found with "/usr/local/kong/ssl/kongcert.pem"
2019/07/29 18:30:51 [debug] KONG_DNS_NO_SYNC ENV found with "off"
2019/07/29 18:30:51 [debug] KONG_CASSANDRA_KEYSPACE ENV found with "kong_dev"
2019/07/29 18:30:51 [debug] KONG_DB_CACHE_TTL ENV found with "0"
2019/07/29 18:30:51 [debug] KONG_CLIENT_SSL ENV found with "off"
2019/07/29 18:30:51 [debug] KONG_PLUGINS ENV found with "kong-plugin-oauth,zipkin,kong-error-log,kong-oidc-implicit-token,kong-response-size-limiting,request-transformer,kong-service-virtualization,kong-cluster-drain,kong-upstream-jwt,kong-splunk-log,kong-spec-expose,kong-path-based-routing,kong-oidc-multi-idp,correlation-id,oauth2,statsd,jwt,rate-limiting,acl,request-size-limiting,request-termination,cors"
2019/07/29 18:30:51 [debug] admin_access_log = "off"
2019/07/29 18:30:51 [debug] admin_error_log = "/dev/stderr"
2019/07/29 18:30:51 [debug] admin_listen = {"0.0.0.0:8001"}
2019/07/29 18:30:51 [debug] anonymous_reports = false
2019/07/29 18:30:51 [debug] cassandra_consistency = "LOCAL_QUORUM"
2019/07/29 18:30:51 [debug] cassandra_contact_points = {"server8173","server8172","server8175","server8171","server8176","server8180"}
2019/07/29 18:30:51 [debug] cassandra_data_centers = {"dc1:2","dc2:3"}
2019/07/29 18:30:51 [debug] cassandra_keyspace = "kong_dev"
2019/07/29 18:30:51 [debug] cassandra_lb_policy = "RequestDCAwareRoundRobin"
2019/07/29 18:30:51 [debug] cassandra_local_datacenter = "DC2"
2019/07/29 18:30:51 [debug] cassandra_password = "******"
2019/07/29 18:30:51 [debug] cassandra_port = 9042
2019/07/29 18:30:51 [debug] cassandra_repl_factor = 1
2019/07/29 18:30:51 [debug] cassandra_repl_strategy = "SimpleStrategy"
2019/07/29 18:30:51 [debug] cassandra_schema_consensus_timeout = 10000
2019/07/29 18:30:51 [debug] cassandra_ssl = true
2019/07/29 18:30:51 [debug] cassandra_ssl_verify = true
2019/07/29 18:30:51 [debug] cassandra_timeout = 5000
2019/07/29 18:30:51 [debug] cassandra_username = "dba_user"
2019/07/29 18:30:51 [debug] client_body_buffer_size = "50m"
2019/07/29 18:30:51 [debug] client_max_body_size = "100m"
2019/07/29 18:30:51 [debug] client_ssl = false
2019/07/29 18:30:51 [debug] database = "cassandra"
2019/07/29 18:30:51 [debug] db_cache_ttl = 0
2019/07/29 18:30:51 [debug] db_cache_warmup_entities = {"services","plugins","consumers"}
2019/07/29 18:30:51 [debug] db_resurrect_ttl = 30
2019/07/29 18:30:51 [debug] db_update_frequency = 5
2019/07/29 18:30:51 [debug] db_update_propagation = 5
2019/07/29 18:30:51 [debug] dns_error_ttl = 1
2019/07/29 18:30:51 [debug] dns_hostsfile = "/etc/hosts"
2019/07/29 18:30:51 [debug] dns_no_sync = false
2019/07/29 18:30:51 [debug] dns_not_found_ttl = 30
2019/07/29 18:30:51 [debug] dns_order = {"LAST","SRV","A","CNAME"}
2019/07/29 18:30:51 [debug] dns_resolver = {}
2019/07/29 18:30:51 [debug] dns_stale_ttl = 4
2019/07/29 18:30:51 [debug] error_default_type = "text/plain"
2019/07/29 18:30:51 [debug] headers = {"latency_tokens"}
2019/07/29 18:30:51 [debug] log_level = "notice"
2019/07/29 18:30:51 [debug] lua_package_cpath = ""
2019/07/29 18:30:51 [debug] lua_package_path = "./?.lua;./?/init.lua;"
2019/07/29 18:30:51 [debug] lua_socket_pool_size = 30
2019/07/29 18:30:51 [debug] lua_ssl_trusted_certificate = "/usr/local/kong/ssl/kongcert.pem"
2019/07/29 18:30:51 [debug] lua_ssl_verify_depth = 3
2019/07/29 18:30:51 [debug] mem_cache_size = "1024m"
2019/07/29 18:30:51 [debug] nginx_admin_directives = {}
2019/07/29 18:30:51 [debug] nginx_daemon = "on"
2019/07/29 18:30:51 [debug] nginx_http_directives = {}
2019/07/29 18:30:51 [debug] nginx_optimizations = true
2019/07/29 18:30:51 [debug] nginx_proxy_directives = {}
2019/07/29 18:30:51 [debug] nginx_sproxy_directives = {}
2019/07/29 18:30:51 [debug] nginx_stream_directives = {}
2019/07/29 18:30:51 [debug] nginx_user = "nobody nobody"
2019/07/29 18:30:51 [debug] nginx_worker_processes = "auto"
2019/07/29 18:30:51 [debug] origins = {}
2019/07/29 18:30:51 [debug] pg_database = "kong"
2019/07/29 18:30:51 [debug] pg_host = "127.0.0.1"
2019/07/29 18:30:51 [debug] pg_max_concurrent_queries = 0
2019/07/29 18:30:51 [debug] pg_port = 5432
2019/07/29 18:30:51 [debug] pg_semaphore_timeout = 60000
2019/07/29 18:30:51 [debug] pg_ssl = false
2019/07/29 18:30:51 [debug] pg_ssl_verify = false
2019/07/29 18:30:51 [debug] pg_timeout = 5000
2019/07/29 18:30:51 [debug] pg_user = "kong"
2019/07/29 18:30:51 [debug] plugins = {"kong-plugin-oauth","zipkin","kong-error-log","kong-oidc-implicit-token","kong-response-size-limiting","request-transformer","kong-service-virtualization","kong-cluster-drain","kong-upstream-jwt","kong-splunk-log","kong-spec-expose","kong-path-based-routing","kong-oidc-multi-idp","correlation-id","oauth2","statsd","jwt","rate-limiting","acl","request-size-limiting","request-termination","cors"}
2019/07/29 18:30:51 [debug] prefix = "/usr/local/kong/"
2019/07/29 18:30:51 [debug] proxy_access_log = "off"
2019/07/29 18:30:51 [debug] proxy_error_log = "/dev/stderr"
2019/07/29 18:30:51 [debug] proxy_listen = {"0.0.0.0:8000","0.0.0.0:8443 ssl http2"}
2019/07/29 18:30:51 [debug] real_ip_header = "X-Forwarded-For"
2019/07/29 18:30:51 [debug] real_ip_recursive = "on"
2019/07/29 18:30:51 [debug] router_consistency = "eventual"
2019/07/29 18:30:51 [debug] ssl_cert = "/usr/local/kong/ssl/kongcert.crt"
2019/07/29 18:30:51 [debug] ssl_cert_key = "/usr/local/kong/ssl/kongprivatekey.key"
2019/07/29 18:30:51 [debug] ssl_cipher_suite = "modern"
2019/07/29 18:30:51 [debug] ssl_ciphers = "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256"
2019/07/29 18:30:51 [debug] stream_listen = {"off"}
2019/07/29 18:30:51 [debug] trusted_ips = {"0.0.0.0/0","::/0"}
2019/07/29 18:30:51 [debug] upstream_keepalive = 60
2019/07/29 18:30:51 [verbose] prefix in use: /usr/local/kong
2019/07/29 18:30:51 [debug] resolved Cassandra contact point 'server8173' to: 10.87.xxx.xx
2019/07/29 18:30:51 [debug] resolved Cassandra contact point 'server8172' to: 10.87.xxx.xx
2019/07/29 18:30:51 [debug] resolved Cassandra contact point 'server8175' to: 10.87.xxx.xx
2019/07/29 18:30:51 [debug] resolved Cassandra contact point 'server8171' to: 10.86.xx.xx
2019/07/29 18:30:51 [debug] resolved Cassandra contact point 'server8176' to: 10.86.xxx.xx
2019/07/29 18:30:51 [debug] resolved Cassandra contact point 'server8180' to: 10.86.xx.xx
2019/07/29 18:30:51 [debug] loading subsystems migrations...
2019/07/29 18:30:51 [verbose] retrieving keyspace schema state...
2019/07/29 18:30:51 [verbose] schema state retrieved
2019/07/29 18:30:51 [debug] loading subsystems migrations...
2019/07/29 18:30:51 [verbose] retrieving keyspace schema state...
2019/07/29 18:30:51 [verbose] schema state retrieved
2019/07/29 18:30:51 [info] No pending migrations to finish
2019/07/29 18:30:52 [info] aborted: another node is performing database changes
```

```
dba_user@cqlsh:kong_dev> select * from schema_meta;

 key         | subsystem             | executed                                                                        | last_executed  | pending
-------------+-----------------------+---------------------------------------------------------------------------------+----------------+---------
 schema_meta |                   acl |                                                {'000_base_acl', '001_14_to_15'} |   001_14_to_15 |    null
 schema_meta |            basic-auth |                                         {'000_base_basic_auth', '001_14_to_15'} |   001_14_to_15 |    null
 schema_meta |                  core | {'000_base', '001_14_to_15', '002_15_to_1', '003_100_to_110', '004_110_to_120'} | 004_110_to_120 |    null
 schema_meta |             hmac-auth |                                          {'000_base_hmac_auth', '001_14_to_15'} |   001_14_to_15 |    null
 schema_meta |                   jwt |                                                {'000_base_jwt', '001_14_to_15'} |   001_14_to_15 |    null
 schema_meta |              key-auth |                                           {'000_base_key_auth', '001_14_to_15'} |   001_14_to_15 |    null
 schema_meta |                oauth2 |                             {'000_base_oauth2', '001_14_to_15', '002_15_to_10'} |   002_15_to_10 |    null
 schema_meta |         rate-limiting |     {'000_base_rate_limiting', '001_14_to_15', '002_15_to_10', '003_10_to_112'} |  003_10_to_112 |    null
 schema_meta | response-ratelimiting |             {'000_base_response_rate_limiting', '001_14_to_15', '002_15_to_10'} |   002_15_to_10 |    null
```

### Additional Details & Logs

- Kong version 1.2.1 leveraging C*
- Operating system Alpine Docker Kong

**Comment by jeremyjpj0916** [comment]
Another interesting bit is when I try to query this specific table I get an error, maybe related?

```
dba@cqlsh:kong_stage> select * from cluster_events;
ReadFailure: Error from server: code=1300 [Replica(s) failed to execute read] message="Operation failed - received 0 responses and 1 failures" info={'failures': 1, 'received_responses': 0, 'required_responses': 1, 'consistency': 'ONE'}
```

Yet nodetool status says all nodes are healthy and Kong seems to be functioning fine and queryable with all other tables in the keyspace.

**Comment by jeremyjpj0916** [comment]
Further debugging into the C* logs had this:
```
DEBUG [pool-1-thread-4] 2019-07-31 00:02:10,662 StorageProxy.java:2371 - Schemas are in agreement.
DEBUG [pool-1-thread-3] 2019-07-31 00:02:20,719 StorageProxy.java:2371 - Schemas are in agreement.
DEBUG [pool-1-thread-1] 2019-07-31 00:02:30,540 StorageProxy.java:2371 - Schemas are in agreement.
DEBUG [pool-1-thread-5] 2019-07-31 00:02:40,774 StorageProxy.java:2371 - Schemas are in agreement.
DEBUG [pool-1-thread-2] 2019-07-31 00:02:50,609 StorageProxy.java:2371 - Schemas are in agreement.
WARN  [ReadStage-2] 2019-07-31 00:02:59,510 ReadCommand.java:569 - Read 0 live rows and 100001 tombstone cells for query SELECT * FROM kong_stage.cluster_events LIMIT 100 (see tombstone_warn_threshold)
ERROR [ReadStage-2] 2019-07-31 00:02:59,510 StorageProxy.java:1909 - Scanned over 100001 tombstones during query 'SELECT * FROM kong_stage.cluster_events LIMIT 100' (last scanned row partion key was ((invalidations), 2019-07-29 01:42-0500, 2d272dde-b8e9-4384-a33d-baf2b2ce2042, 7440f955-7140-4d67-bec7-ea7e5ddb1c6c)); query aborted
DEBUG [Native-Transport-Requests-1] 2019-07-31 00:02:59,510 ReadCallback.java:132 - Failed; received 0 of 1 responses
DEBUG [pool-1-thread-4] 2019-07-31 00:03:00,677 StorageProxy.java:2371 - Schemas are in agreement.
DEBUG [pool-1-thread-3] 2019-07-31 00:03:10,696 StorageProxy.java:2371 - Schemas are in agreement.
DEBUG [pool-1-thread-1] 2019-07-31 00:03:20,726 StorageProxy.java:2371 - Schemas are in agreement.
DEBUG [pool-1-thread-5] 2019-07-31 00:03:30,682 StorageProxy.java:2371 - Schemas are in agreement.
DEBUG [pool-1-thread-2] 2019-07-31 00:03:40,640 StorageProxy.java:2371 - Schemas are in agreement.
DEBUG [pool-1-thread-4] 2019-07-31 00:03:50,915 StorageProxy.java:2371 - Schemas are in agreement.
DEBUG [pool-1-thread-3] 2019-07-31 00:04:00,672 StorageProxy.java:2371 - Schemas are in agreement.
WARN  [ReadStage-2] 2019-07-31 00:04:08,945 ReadCommand.java:569 - Read 62 live rows and 3165 tombstone cells for query SELECT * FROM kong_stage.services WHERE partition = services LIMIT 1000 (see tombstone_warn_threshold)
WARN  [ReadStage-2] 2019-07-31 00:04:08,982 ReadCommand.java:569 - Read 63 live rows and 3169 tombstone cells for query SELECT * FROM kong_stage.routes WHERE partition = routes LIMIT 1000 (see tombstone_warn_threshold)
WARN  [ReadStage-2] 2019-07-31 00:04:08,983 ReadCommand.java:569 - Read 63 live rows and 3169 tombstone cells for query SELECT * FROM kong_stage.routes WHERE partition = routes LIMIT 1000 (see tombstone_warn_threshold)
```

Very specifically this bit in the above ^:
```
WARN  [ReadStage-2] 2019-07-31 00:02:59,510 ReadCommand.java:569 - Read 0 live rows and 100001 tombstone cells for query SELECT * FROM kong_stage.cluster_events LIMIT 100 (see tombstone_warn_threshold)
ERROR [ReadStage-2] 2019-07-31 00:02:59,510 StorageProxy.java:1909 - Scanned over 100001 tombstones during query 'SELECT * FROM kong_stage.cluster_events LIMIT 100' (last scanned row partion key was ((invalidations), 2019-07-29 01:42-0500, 2d272dde-b8e9-4384-a33d-baf2b2ce2042, 7440f955-7140-4d67-bec7-ea7e5ddb1c6c)); query aborted
DEBUG [Native-Transport-Requests-1] 2019-07-31 00:02:59,510 ReadCallback.java:132 - Failed; received 0 of 1 responses
```

C* 3.11.1 in these environments.

**Comment by jeremyjpj0916** [comment]
Apparently this may fix it?

https://medium.com/@eun9882/cassandra-trouble-shooting-read-failed-53c2024b048b (have to translate some of the page and it says this:
```
Problem
select 실행시 read fail
Error syntax
ReadFailure: Error from server: code=1300 [Replica(s) failed to execute read] message=”Operation failed — received 0 responses and 1 failures” info={‘failures’: 1, ‘received_responses’: 0, ‘required_responses’: 1, ‘consistency’: ‘ONE’}
Read 897 live rows and 17872 tombstone cells
solution
Clean up tombstone
ALTER TABLE thunnus.metric_diag WITH GC_GRACE_SECONDS = 0;
cause
exceeding the tombstone_failure_threshold. It defaults to 100'000
increase the value in the cassandra.yaml or
clean up tombstones -> flush all deleted marks
```

**Comment by jeremyjpj0916** [comment]
I am thinking we can follow this:
https://support.datastax.com/hc/en-us/articles/360001108603-FAQ-Why-is-forcing-major-compaction-on-a-table-not-ideal-

So something like:
```
nodetool compact --split-output -- kong_stage cluster_events
```

Should fix this.

Here is the table info too:
```
CREATE TABLE kong_stage.cluster_events (
    channel text,
    at timestamp,
    node_id uuid,
    id uuid,
    data text,
    nbf timestamp,
    PRIMARY KEY (channel, at, node_id, id)
) WITH CLUSTERING ORDER BY (at ASC, node_id ASC, id ASC)
    AND bloom_filter_fp_chance = 0.01
    AND caching = {'keys': 'ALL', 'rows_per_partition': 'NONE'}
    AND comment = 'Kong cluster events broadcasting and polling'
    AND compaction = {'class': 'org.apache.cassandra.db.compaction.SizeTieredCompactionStrategy', 'max_threshold': '32', 'min_threshold': '4'}
    AND compression = {'chunk_length_in_kb': '64', 'class': 'org.apache.cassandra.io.compress.LZ4Compressor'}
    AND crc_check_chance = 1.0
    AND dclocal_read_repair_chance = 0.1
    AND default_time_to_live = 86400
    AND gc_grace_seconds = 864000
    AND max_index_interval = 2048
    AND memtable_flush_period_in_ms = 0
    AND min_index_interval = 128
    AND read_repair_chance = 0.0
    AND speculative_retry = '99PERCENTILE';
```

**Comment by jeremyjpj0916** [comment]
Urg.. well one stage environment I was able to temp fix by increasing the tombstone_failure_threshold from the 100,000 default to 200,000 . But our main Stage must have over 200+ cause I got this:
```
ERROR [ReadStage-1] 2019-07-31 01:18:29,043 StorageProxy.java:1906 - Scanned over 200001 tombstones during query 'SELECT * FROM kong_stage.cluster_events LIMIT 100' (last scanned row partion key was ((invalidations), 2019-07-24 21:14-0500, 6bc476af-7975-4e14-add3-a86e923a3e6d, 9485c3a5-75d4-456d-9350-e3fd70c294c4)); query aborted
DEBUG [Native-Transport-Requests-2] 2019-07-31 01:18:29,043 ReadCallback.java:132 - Failed; received 0 of 1 responses
DEBUG [pool-1-thread-5] 2019-07-31 01:18:32,600 StorageProxy.java:2368 - Schemas are in agreement.
```

I think my only choice is to adjust the gc_grace_seconds to something short like 60 seconds to get agressive and see what happens.

Might be worth Kong revisiting this table and going with a different strategy/settings since cluster_events seems to me a highly intensive table?
https://www.manniwood.com/2017_11_05/aggressive_cassandra_compaction.html

What is really weird is why is this happening now all of the sudden trying to go from 1.0.3 to 1.2.1... hmmm.

**Comment by jeremyjpj0916** [comment]
So in the finish example above, we must be hitting here:
https://github.com/Kong/kong/blob/19f4fe64c7278ce57696f3600b204744d07b5a5d/kong/cmd/utils/migrations.lua#L197

And then ok is ```nil``` or ```false``` at that point depending on how that db call returns(I didn't dig in), yet based on the log output we also never hit here right?
https://github.com/Kong/kong/blob/19f4fe64c7278ce57696f3600b204744d07b5a5d/kong/cmd/utils/migrations.lua#L232

Because if there was an error wouldn't ```error(err)``` give some extra detail to any issues it sees?

So then based on log I left at the top we drop into here:
https://github.com/Kong/kong/blob/19f4fe64c7278ce57696f3600b204744d07b5a5d/kong/cmd/utils/migrations.lua#L236

**Comment by jeremyjpj0916** [comment]
Figured this out out too folks, since I run a modified concurrency.lua (as seen and discussed here: https://github.com/Kong/kong/issues/4769#issuecomment-516069861 ) I decided to take it out at runtime and just run with the default one that doesn't do db keepalive in that portion of the code and that indeed caused the false positive message to stop coming through. 

I am sure you folks know why keepalive to the db in concurrency would likely produce the behavior shown above that in the db:cluster_mutex calls in the migration cli operations. At the end of the day it seems harmless and I can't run Kong without that concurrency hack without paging issues during router/plugin rebuilds so closing this as its not relevant to the main Kong branch! 

My guess is the tweak causes it to hit in one of these 3 spots:
https://github.com/Kong/kong/blob/19f4fe64c7278ce57696f3600b204744d07b5a5d/kong/db/init.lua#L298

https://github.com/Kong/kong/blob/19f4fe64c7278ce57696f3600b204744d07b5a5d/kong/db/init.lua#L312

https://github.com/Kong/kong/blob/19f4fe64c7278ce57696f3600b204744d07b5a5d/kong/db/init.lua#L334

Sorry again! Glad I had time to work through it some today as I wanted to feel better about this not being a major concern when upgrading prod to 1.2.1 .
